File: Motocitizen/src/motocitizen/Activity/DetailHistoryFragment.java
package motocitizen.Activity;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import motocitizen.app.general.Accident;
import motocitizen.app.general.AccidentHistory;
import motocitizen.main.R;

public class DetailHistoryFragment extends AccidentDetailsFragments {

    private OnFragmentInteractionListener mListener;

    private View mcDetLogContent;

    public static DetailHistoryFragment newInstance(int accID, String userName) {
        DetailHistoryFragment fragment = new DetailHistoryFragment();
        Bundle args = new Bundle();
        args.putInt(ACCIDENT_ID, accID);
        args.putString(USER_NAME, userName);
        fragment.setArguments(args);
        return fragment;
    }

    public DetailHistoryFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View viewMain = inflater.inflate(R.layout.fragment_detail_history, container, false);
        mcDetLogContent = viewMain.findViewById(R.id.mc_det_log_content);

        update();
        return viewMain;
    }

    protected void update() {
        Accident accident = ((AccidentDetailsActivity) getActivity()).getCurrentPoint();

        ViewGroup logView = (ViewGroup) mcDetLogContent;
        logView.removeAllViews();
        logView.addView(AccidentHistory.createHeader(getActivity()));
        for (int i : accident.getSortedHistoryKeys()) {
            logView.addView(accident.history.get(i).createRow(getActivity(), userName));
        }
    }

    public void notifyDataSetChanged() {
        update();
    }
}

File: Motocitizen/src/motocitizen/ui/fragments/DetailHistoryFragment.kt
package motocitizen.ui.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.LinearLayout
import motocitizen.content.Content
import motocitizen.content.accident.Accident
import motocitizen.main.R
import motocitizen.ui.activity.AccidentDetailsActivity.Companion.ACCIDENT_ID_KEY
import motocitizen.ui.rows.HistoryRow

class DetailHistoryFragment : FragmentForAccident() {
    companion object {
        private val ROOT_LAYOUT = R.layout.fragment_detail_history
        private val CONTENT_VIEW = R.id.details_log_content
    }

    private lateinit var rootView: View
    private val logContent: LinearLayout by lazy { rootView.findViewById(CONTENT_VIEW)!! }
    private lateinit var accident: Accident

    override fun setAccident(accident: Accident) {
        this.accident = accident
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        rootView = inflater.inflate(ROOT_LAYOUT, container, false)
        redrawHistory()
        return rootView
    }

    private fun redrawHistory() {
        logContent.removeAllViews()
        accident.history.forEach {
            try {
                logContent.addView(HistoryRow(requireActivity(), it))
            } catch (e: IllegalArgumentException) {
                //todo fuckup
            }
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putInt(ACCIDENT_ID_KEY, accident.id)
        super.onSaveInstanceState(outState)

    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        if (savedInstanceState == null) return
        accident = Content[savedInstanceState.getInt(ACCIDENT_ID_KEY)]!!
    }
}